app/packages/core/src/components/Actions/ActionsRow.tsx

```diff
@@ -301,7 +300,18 @@ const SaveFilters = () => {
     []
   );
 
-  return hasFiltersValue ? (
+  const hasFiltersValue = useRecoilValue(fos.hasFilters(false));
+
+  const extendedSelectionList = useRecoilValue(fos.extendedSelection);
+  const isExtendedSelectionOn =
+    extendedSelectionList && extendedSelectionList.length > 0;
+
+  const selectedSampleSet = useRecoilValue(fos.selectedSamples);
+
+  const shouldToggleBookMarkIconOn =
+    hasFiltersValue || isExtendedSelectionOn || selectedSampleSet.size > 0;
```

Comment:
It might be good to create a selector for this to isolate the logic and reduce renders. In `@fiftyone/state/src/recoil/selectors.ts` maybe